Split core logic into a separate packable library

achampto opened the pull request:
This pulls the core extractor logic into a separate library and allows for the creation of a nuget package for that core logic so that it can be consumed by other applications. I've attempted to change as little logic as possible. The largest change is updating the hero types and multipliers json files to be embedded resources so that they're easily readable when consuming this as a nuget package.

LukeCroteau commented:
@achampto 
There are quite a few changes in here, great work!

Question: Did you ensure the GenerateClients.ps1 script creates the nswag information correctly automatically?
My gut feeling is that this has not been taken into account. The AccountDumpClient.cs is automatically generated using this script, along with the Website side client.ts file. 

It looks like you've manually extracted the relevant bits out, breaking some of that functionality.

I'd suggest that we probably want to preserve this. Any thoughts?

achampto commented:
@LukeCroteau I wasn't aware of that GenerateClients script. I moved the entire AccountDumpClient into the Core library and updated the script to point to that location. I don't love having the HttpClient inside the Core library though so there's opportunity to do something different there in the future, but for now it creates some separation while keeping the full tooling chain working

LukeCroteau commented:
@achampto beautiful work! Thank you!
